cmd/deploy/deploy.go

```diff
@@ -278,6 +281,27 @@ func (dc *deployCommand) runDeploy(ctx context.Context, cwd string, opts *Option
 	oktetoLog.Debugf("starting server on %d", dc.proxy.GetPort())
 	dc.proxy.Start()
 
+	oktetoLog.LogIntoBuffer("Deploying app '%s'...", opts.Name)
+	data := &app.CfgData{
+		Repository: os.Getenv(model.GithubRepositoryEnvVar),
+		Branch:     os.Getenv(model.OktetoGitBranchEnvVar),
+		Filename:   opts.Manifest.Filename,
+		Status:     app.ProgressingStatus,
+		Manifest:   opts.Manifest.Manifest,
+		Icon:       opts.Manifest.Icon,
+	}
+
+	k8sCfg := kubeconfig.Get(config.GetKubeconfigPath())
+	c, _, err := dc.k8sClientProvider.Provide(k8sCfg)
+	if err != nil {
+		return err
+	}
+	cfg := app.TranslateConfigMap(ctx, opts.Name, opts.Manifest.Namespace, data, c)
+
+	if err := configmaps.Deploy(ctx, cfg, opts.Manifest.Namespace, c); err != nil {
```

Comment:
we should get the value and modify what's needed to be compatible with pipelines running at that time. if the configmap has changed when updating it, we should read it again.